src/main/java/com/google/step/servlets/UserServlet.java

```diff
@@ -56,6 +59,33 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
         new ArrayList<>(); // followManager.getRestaurantsFollowedByUser(id);
 
     String json = JsonFormatter.getUserJson(user, deals, following, followers, tags, restaurants);
+    response.setContentType("application/json");
     response.getWriter().println(json);
   }
+
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    if (!userService.isUserLoggedIn()) {
+      response.setStatus(HttpServletResponse.SC_BAD_REQUEST);
+      return;
+    }
+    String userEmail = userService.getCurrentUser().getEmail();
+
+    String username = (String) request.getParameter("username");
+    String bio = (String) request.getParameter("bio");
+    String email = (String) request.getParameter("email");
+
+    if (!userEmail.equals(email)) {
+      // Inconsistent request with login status
+      response.setStatus(HttpServletResponse.SC_BAD_REQUEST);
+      return;
+    }
+
+    User user = userManager.readOrCreateUserByEmail(userEmail);
```

Comment:
the id in the url /users/<id> is not used at all. I think you should get the user from the id, and make sure that only the logged in user can edit his own profile

Reply:
oh i was thinking that patch request does not have an id parameter. maybe I should include id as well